tests/test_database_manager.py

```diff
@@ -0,0 +1,179 @@
+
+        assert (await dbm_fx.select_rows("testtableselect", "AND", {"string1": "thest"}))[0][0]\
+            == ("thest",)[0]
+
+    async def test_create_table(self, dbm_fx: DatabaseManager):
+        if await dbm_fx.has_table("testtablecreate"):
+            await dbm_fx.drop_table("testtablecreate")
+        if not await dbm_fx.has_table("testtablecreate"):
+            await dbm_fx.create_table("testtablecreate", {"string1": "VARCHAR UNIQUE"})
+        with pytest.raises(ValueError):
+            await dbm_fx.create_table("testtablecreate", {"test": "VARCHAR UNIQUE"})
+
+    async def test_drop_table(self, dbm_fx: DatabaseManager):
+        if not await dbm_fx.has_table("testtabledrop"):
+            await dbm_fx.create_table("testtabledrop", {"string1": "TIMESTAMP UNIQUE"})
+        assert await dbm_fx.has_table("testtabledrop")
+        await dbm_fx.drop_table("testtabledrop")
+        assert not await dbm_fx.has_table("testtabledrop")
+
+    async def test_insert_row(self, dbm_fx: DatabaseManager):
+        if not await dbm_fx.has_table("testtableinsert"):
+            await dbm_fx.create_table("testtableinsert", {"string1": "VARCHAR UNIQUE"})
+        await dbm_fx.insert_row("testtableinsert", {"string1": "test"}, ("string1",))
+        tmp = await dbm_fx.select_rows("testtableinsert", "AND", {"string1": "test"})
+        assert tmp[0][0] == ("test",)[0]
+
+    async def test_update_row(self, dbm_fx: DatabaseManager):
+        if not await dbm_fx.has_table("testtableupdate"):
+            await dbm_fx.create_table("testtableupdate", {"string1": "VARCHAR UNIQUE"})
+        await dbm_fx.insert_row("testtableupdate", {"string1": "thest"}, ("string1",))
+        await dbm_fx.update_row("testtableupdate", "AND", {"string1": "FizzBuzz"},
+                                {"string1": "thest"})
+
+        tmp = await dbm_fx.select_rows("testtableupdate", "AND", {"string1": "FizzBuzz"})
+        assert tmp[0][0] == "FizzBuzz"
+
+    async def test_delete_row(self, dbm_fx: DatabaseManager):
+        if not await dbm_fx.has_table("testtabledelete"):
+            await dbm_fx.create_table("testtabledelete", {"string1": "VARCHAR UNIQUE"})
+        await dbm_fx.insert_row("testtabledelete", {"string1": "thest"}, ("string1",))
+        await dbm_fx.delete_row("testtabledelete", "AND", {"string1": "thest"})
+        tmp = await dbm_fx.select_rows("testtabledelete", "AND", {"string1": "thest"})
+        assert len(tmp) == 0
+
+    async def test_double_dash_checks(self, dbm_fx: DatabaseManager):
+        with pytest.raises(ValueError):
+            await dbm_fx.delete_row("testtabledelete", "AND", {"test": "OR TRUE; -- "})
+            await dbm_fx.update_row("testtableupdate", "AND", {"string1": " OR FALSE; -- "})
+            await dbm_fx.insert_row("testtableinsert", {"string1": "stuff"}, ("string1",))
+            await dbm_fx.select_rows("testtableselect", "AND", {"test; -- ": "stuff"})
+

```

Comment:
only line 115 executes. presuming each of these calls is supposed to raise a valueerror.